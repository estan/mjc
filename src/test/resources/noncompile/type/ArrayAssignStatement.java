class ArrayAssignStatementTest {
    public static void main(String[] args) {
        int i;
        int[] ia;
        boolean b;
        A cA;
        B cB;

        //ia[i] = i;  // OK!
        ia[ia] = i;   // INVALID_INDEX_TYPE
        ia[b] = i;    // INVALID_INDEX_TYPE
        ia[cA] = i;   // INVALID_INDEX_TYPE

        A[i] = i; // EXPECTED_VARIABLE_GOT_CLASS
        U[i] = i; // UNDECLARED_IDENTIFIER

        ia[i] = ia; // INVALID_ASSIGNMENT
        ia[i] = b;  // INVALID_ASSIGNMENT
        ia[i] = cA; // INVALID_ASSIGNMENT

        i[i] = i;  // NOT_ARRAY_TYPE
        b[i] = i;  // NOT_ARRAY_TYPE
        cA[i] = i; // NOT_ARRAY_TYPE
    }
}
class A {}
class B {}
