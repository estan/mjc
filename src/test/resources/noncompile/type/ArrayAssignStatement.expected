INVALID_INDEX_TYPE
INVALID_INDEX_TYPE
INVALID_INDEX_TYPE
EXPECTED_VARIABLE_GOT_CLASS
UNDECLARED_IDENTIFIER
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
INVALID_ASSIGNMENT
NOT_ARRAY_TYPE
NOT_ARRAY_TYPE
NOT_ARRAY_TYPE
